turbopack/crates/turbo-tasks-backend/src/backend/mod.rs

```diff
@@ -795,6 +884,9 @@ impl<B: BackingStorage> TurboTasksBackendInner<B> {
         task_id: TaskId,
         turbo_tasks: &dyn TurboTasksBackendApi<TurboTasksBackend<B>>,
     ) {
+        if !self.should_track_dependencies() {
+            panic!("Dependency tracking is disabled so invalidation is not allowed");
```

Comment:
Should this (and related panics) be an `unreachable!`?

Reply:
It's not unreachable. you would hit this panic when you disable dependency tracking, but enable watching on filesystem, or try to change state.